system-x/services/splunk/src/main/java/software/tnb/splunk/resource/openshift/OpenshiftSplunk.java

```diff
@@ -38,14 +39,42 @@
 @AutoService(Splunk.class)
 public class OpenshiftSplunk extends Splunk implements ReusableOpenshiftDeployable {
     private static final Logger LOG = LoggerFactory.getLogger(OpenshiftSplunk.class);
-    public static final String ROUTE_NAME = "api";
-    public static final int API_PORT = 443;
+    private static final String CRD_API = "v4";
+    private static final String ROUTE_NAME = "api";
     private static List<HasMetadata> createdResources;
     private CustomResourceDefinitionContext crdContext;
     private Route apiRoute;
-
     private String sccName;
 
+    public OpenshiftSplunk() {
+        getConfiguration().protocol(SplunkProtocol.HTTPS);
+    }
+
+    private Map<String, Object> getSplunkCr() {
+        if (getConfiguration().getProtocol().equals(SplunkProtocol.HTTPS)) {
+            return Map.of(
+                "apiVersion", "enterprise.splunk.com/" + CRD_API,
+                "kind", "Standalone",
+                "metadata", Map.of(
+                    "name", "s1")
+            );
+        } else {
+            return Map.of(
```

Comment:
can be simplified, since this just adds one more field into the map:

```
Map config = [apiVersion, kind, metadata]
if (protocol == http) {
    config.put(spec)
}
```